core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
```

Comment:
Not sure about users calling `isEmpty`. I think it's more likely that users will ignore this property or search for annotations that their logic depends on (`annotations.filterIsInstance<UserAnnotation>()`).